src/lib.rs

```diff
@@ -173,28 +293,17 @@ use SizeClass::*;
 // TODO: Find a way to test that this actually does what it claims; that is,
 // that it boxes large values and copies small ones.
 #[repr(transparent)]
-pub struct Stowaway<T> {
-    // TODO: Reimplemnt this as union, once we can have non-copy fields in a
+pub struct Stowaway<T: Stowable> {
+    // TODO: Reimplement this as union, once we can have non-copy fields in a
```

Comment:
You can't use a union here, I assume you want
```
union Stowable<T: Stowable> {
    ptr: *mut T,
    value: T,
}
```
But this would make `Stowable` the same size as `T` when `sizeof(T) > sizeof(ptr)`, which is not what we want.

Reply:
There's an earlier draft of this project- I'm not sure if it's in the git history- where this does actually work, using a trait and an associated type.

Reply:
Oh, I see. I think that would require const generics because you need to choose the layout based on the size and alignment, but that's a nice idea!